File: 0000-0999/0374.GuessNumberHigherorLower.go
/**
 * Forward declaration of guess API.
 * @param  num   your guess
 * @return 	     -1 if num is lower than the guess number
 *			      1 if num is higher than the guess number
 *               otherwise return 0
 * func guess(num int) int;
 */

func guessNumber(n int) int {
	left, right := 1, n
	for left <= right {
		mid := (left + right) / 2
		g := guess(mid)
		if g == 0 {
			return mid
		}
		if g < 0 {
			right = mid - 1
		} else {
			left = mid + 1
		}
	}
	return -1
}

File: 0000-0999/0374.GuessNumberHigherorLower.py
# The guess API is already defined for you.
# @param num, your guess
# @return -1 if my number is lower, 1 if my number is higher, otherwise return 0
# def guess(num):

class Solution(object):
    def guessNumber(self, n):
        """
        :type n: int
        :rtype: int
        """
        left, right = 1, n
        while left <= right:
            mid = left + (right-left)/2
            g = guess(mid)
            if g == 0:
                return mid
            elif g == 1:
                left = mid + 1
            else:
                right = mid - 1
